diff -ura a/meson.build b/meson.build
--- a/meson.build	2025-10-12 07:51:57.984459380 +0000
+++ b/meson.build	2025-10-12 07:52:17.554545664 +0000
@@ -1986,7 +1986,6 @@
 
 check_version_history_py = find_program('tools/check-version-history.py')
 elf2efi_py = find_program('tools/elf2efi.py')
-export_dbus_interfaces_py = find_program('tools/dbus_exporter.py')
 generate_gperfs = find_program('tools/generate-gperfs.py')
 make_autosuspend_rules_py = find_program('tools/make-autosuspend-rules.py')
 make_directive_index_py = find_program('tools/make-directive-index.py')
@@ -2903,14 +2902,6 @@
 alias_target('update-dbus-docs', update_dbus_docs)
 alias_target('update-man-rules', update_man_rules)
 
-if not meson.is_cross_build()
-        custom_target(
-                'export-dbus-interfaces',
-                output : fs.name(dbus_interfaces_dir),
-                install : dbus_interfaces_dir != 'no',
-                install_dir : fs.parent(dbus_interfaces_dir),
-                command : [export_dbus_interfaces_py, '@OUTPUT@', dbus_programs])
-endif
 
 meson_extract_unit_files = find_program('tools/meson-extract-unit-files.py')
 custom_target('installed-unit-files.txt',
