diff -ura a/login-utils/login.c b/login-utils/login.c
--- a/login-utils/login.c	2025-09-14 19:38:17.454000000 +0000
+++ b/login-utils/login.c	2025-09-14 19:39:05.492000000 +0000
@@ -1579,7 +1579,7 @@
 	child_argv[child_argc++] = NULL;
 
 	/* http://www.linux-pam.org/Linux-PAM-html/adg-interface-by-app-expected.html#adg-pam_end */
-	(void) pam_end(cxt.pamh, PAM_SUCCESS|PAM_DATA_SILENT);
+	(void) pam_end(cxt.pamh, PAM_SUCCESS|PAM_SILENT);
 
 	execvp(child_argv[0], child_argv + 1);
 
diff -ura a/login-utils/su-common.c b/login-utils/su-common.c
--- a/login-utils/su-common.c	2025-09-14 19:38:17.454000000 +0000
+++ b/login-utils/su-common.c	2025-09-14 19:40:19.572000000 +0000
@@ -1259,7 +1259,7 @@
 		warn(_("warning: cannot change directory to %s"), su->pwd->pw_dir);
 
 	/* http://www.linux-pam.org/Linux-PAM-html/adg-interface-by-app-expected.html#adg-pam_end */
-	(void) pam_end(su->pamh, PAM_SUCCESS|PAM_DATA_SILENT);
+	(void) pam_end(su->pamh, PAM_SUCCESS|PAM_SILENT);
 
 	if (shell)
 		run_shell(su, shell, command, argv + optind, max(0, argc - optind));
