diff -ura a/meson.build b/meson.build
--- a/meson.build	2025-09-28 08:21:38.064965195 +0000
+++ b/meson.build	2025-09-28 08:21:47.260073855 +0000
@@ -727,10 +727,6 @@
         conf.set_quoted(name, path)
 endforeach
 
-if run_command(ln, '--relative', '--help', check : false).returncode() != 0
-        error('ln does not support --relative (added in coreutils 8.16)')
-endif
-
 #####################################################################
 
 gperf_test_format = '''
