diff -ura a/libpam/meson.build b/libpam/meson.build
--- a/libpam/meson.build	2025-08-02 15:28:49.797687014 +0000
+++ b/libpam/meson.build	2025-08-02 15:29:27.213000000 +0000
@@ -51,7 +51,7 @@
     '-DDEFAULT_MODULE_PATH="@0@/"'.format(securedir),
     '-DLIBPAM_COMPILE',
   ],
-  dependencies: [libpam_internal_dep, libaudit, libeconf, libdl, libintl],
+  dependencies: [libpam_internal_dep, libaudit, libeconf, libintl],
   link_depends: libpam_link_deps,
   link_args: libpam_link_args,
   version: libpam_version,
diff -ura a/meson.build b/meson.build
--- a/meson.build	2025-08-02 15:28:49.823535707 +0000
+++ b/meson.build	2025-08-02 15:28:59.977000000 +0000
@@ -209,7 +209,6 @@
 endif
 
 
-libdl = dependency('dl')
 
 
 if get_option('i18n').disabled()
diff -ura a/tests/meson.build b/tests/meson.build
--- a/tests/meson.build	2025-08-02 15:28:49.799754974 +0000
+++ b/tests/meson.build	2025-08-02 15:29:46.792000000 +0000
@@ -19,9 +19,6 @@
                'pam_start_confdir',
               ]
   tst_deps = [libpam_internal_dep, libpam_dep]
-  if name == 'dlopen'
-    tst_deps += libdl
-  endif
   tst_name = 'tst-' + name
 
   tst_exe = executable(
